### PR TITLE
fixed crash on windows on stop and start again.

### PR DESCRIPTION
**Pull request by FONQRI**
# Pull Request Details

## Checklist

- [ ] I have read the CONTRIBUTING document.
- [x] My code follows the code style of this project. See CODING.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Comment by FONQRI** [comment]
it solved issues explaned in this issue in gnuradio and uhd projects :
gnuradio issue : https://github.com/gnuradio/gnuradio/issues/2351
uhd issue : https://github.com/EttusResearch/uhd/issues/249

**Comment by mbr0wn** [comment]
@FONQRI Thanks a lot for submitting this. At first glance, the change looks good but we need to import it into our internal CI pipeline. I had to change the formatting, the base branch, and the commit message to conform to UHD rules.

File: host/lib/transport/libusb1_zero_copy.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ class libusb_zero_copy_mb : public managed_buffer
     {
         if (wait_for_completion(timeout))
         {
-            if (result.status != LIBUSB_TRANSFER_COMPLETED)
+            if (result.status != LIBUSB_TRANSFER_COMPLETED || result.status != LIBUSB_TRANSFER_CANCELLED)
```
Comment (mbr0wn) [inline host/lib/transport/libusb1_zero_copy.cpp]: This should be && right?

**Comment by mbr0wn** [comment]
@FONQRI Did you run this code? `result.status` can only have one value, so if clause should always throw.

File: host/lib/transport/libusb1_zero_copy.cpp (hunk 1 above)
Comment (FONQRI) [reply host/lib/transport/libusb1_zero_copy.cpp]: Yeah i think it should be && but when i run this code in company it's working in fact it was my teammate code. let me check it. I will tell you when i checked it at saturday. I'm not sure right now maybe something else is wrong.

**Comment by FONQRI** [comment]
> @FONQRI Did you run this code? `result.status` can only have one value, so if clause should always throw.

Yes but let me check it again.

**Comment by FONQRI** [comment]
> @FONQRI Did you run this code? `result.status` can only have one value, so if clause should always throw.

It's right it's || not && because status can only have one value.

**Comment by mbr0wn** [comment]
I tried running this code and it always throws an exception. Look at the clause again:

```cpp
if (result.status != LIBUSB_TRANSFER_COMPLETED || result.status != LIBUSB_TRANSFER_CANCELLED) {}
```

Let's say result.status returns normally, then `result.status == LIBUSB_TRANSFER_COMPLETED`. In that case, `result.status != LIBUSB_TRANSFER_CANCELLED` is true, and we throw an exception. So it needs to be &&.

**Comment by mbr0wn** [comment]
Merged on master.

File: host/lib/transport/libusb1_zero_copy.cpp (hunk 1 above)
Comment (FONQRI) [reply host/lib/transport/libusb1_zero_copy.cpp]: > I tried running this code and it always throws an exception. Look at the clause again:
> 
> ```c++
> if (result.status != LIBUSB_TRANSFER_COMPLETED || result.status != LIBUSB_TRANSFER_CANCELLED) {}
> ```
> 
> Let's say result.status returns normally, then `result.status == LIBUSB_TRANSFER_COMPLETED`. In that case, `result.status != LIBUSB_TRANSFER_CANCELLED` is true, and we throw an exception. So it needs to be &&.

Yes i got it.It was distracting.

**Pull request closed without merging**